liiatools/datasets/s903/s903_main_functions.py

```diff
@@ -73,7 +77,17 @@ def cleanfile(input, la_code, la_log_dir, output):
     prep.drop_empty_rows(input, input)
 
     # Configuration
-    config = clean_config.Config()
+    try:
+        filename = str(Path(input).resolve().stem)
+        year = check_year(filename)
+    except (AttributeError, ValueError):
+        save_year_error(input, la_log_dir)
+        return
+    if year < f"{datetime.now().year - 6}" or  year > f"{datetime.now().year + 1}":
```

Comment:
I would turn this into it's own function and put that function in the common.py module. As it will likely be reused for the other datasets and this will allow you to create a unit test for it. I would also probably include a parameter that relates to number of years, so it can be adjusted for different use cases that have different data retention policies

Reply:
We also want to include a cut-off when new data becomes available, previously we have chosen June so should stick with this. Similar to the remove_old_data() function here: [https://github.com/SocialFinanceDigitalLabs/liia-tools/blob/main/liiatools/datasets/s903/lds_ssda903_la_agg/process.py](url). So currently we would run the process with data from 2017 onwards (2016/17 return), when we hit July we only want to use 2018 onwards (2017/18 return) as the previous return will have been produced over 6 years ago